From 9ee021b17ee24693579155522a95d4c1c5704123 Mon Sep 17 00:00:00 2001
From: Fabrizio Castro <fabrizio.castro.jz@renesas.com>
Date: Tue, 19 Jul 2022 11:41:06 +0100
Subject: [PATCH] arm64: dts: renesas: barcode-poc: Add BT support

The Barcode-PoC comes with a PMOD connector. This patch adds
device tree support for the DA14531EVZ PMOD BT device.

Signed-off-by: Fabrizio Castro <fabrizio.castro.jz@renesas.com>
---
 .../renesas/barcode-poc-da14531evz-pmod.dtsi  | 34 +++++++++++++++++++
 .../dts/renesas/r9a07g044l2-barcode-poc.dts   |  1 +
 2 files changed, 35 insertions(+)
 create mode 100644 arch/arm64/boot/dts/renesas/barcode-poc-da14531evz-pmod.dtsi

diff --git a/arch/arm64/boot/dts/renesas/barcode-poc-da14531evz-pmod.dtsi b/arch/arm64/boot/dts/renesas/barcode-poc-da14531evz-pmod.dtsi
new file mode 100644
index 000000000000..1650d49b1e50
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/barcode-poc-da14531evz-pmod.dtsi
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for connecting the DA14531EVZ BT PMOD module
+ *
+ * Copyright (C) 2022 Renesas Electronics Corp.
+ */
+
+/ {
+	aliases {
+		serial2 = &scif2;
+	};
+};
+
+&pinctrl {
+	scif2_e_pins: scif2 {
+		pinmux = <RZG2L_PORT_PINMUX(48, 3, 1)>, /* SCIF2_CTS */
+			 <RZG2L_PORT_PINMUX(48, 0, 1)>, /* SCIF2_TXD */
+			 <RZG2L_PORT_PINMUX(48, 1, 1)>, /* SCIF2_RXD */
+			 <RZG2L_PORT_PINMUX(48, 4, 1)>; /* SCIF2_RTS */
+	};
+};
+
+&scif2 {
+	pinctrl-0 = <&scif2_e_pins>;
+	pinctrl-names = "default";
+
+	uart-has-rtscts;
+	status = "okay";
+
+	bluetooth {
+		compatible = "renesas,DA14531";
+		reset-gpios = <&pinctrl RZG2L_GPIO(5, 2) GPIO_ACTIVE_LOW>;
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/r9a07g044l2-barcode-poc.dts b/arch/arm64/boot/dts/renesas/r9a07g044l2-barcode-poc.dts
index a562db813276..2aa86faa3632 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044l2-barcode-poc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g044l2-barcode-poc.dts
@@ -13,6 +13,7 @@
 #include "r9a07g044l2.dtsi"
 #include "rzg2l-smarc-som.dtsi"
 #include "barcode-poc-carrier.dtsi"
+#include "barcode-poc-da14531evz-pmod.dtsi"
 
 / {
 	model = "Barcode PoC based on r9a07g044l2";
-- 
2.34.1

